cmd/bnsd/x/blueaccount/handler.go

```diff
@@ -0,0 +1,519 @@
+package blueaccount
+
+import (
+	"regexp"
+
+	weave "github.com/iov-one/weave"
+	"github.com/iov-one/weave/errors"
+	"github.com/iov-one/weave/gconf"
+	"github.com/iov-one/weave/migration"
+	"github.com/iov-one/weave/orm"
+	"github.com/iov-one/weave/x"
+)
+
+func RegisterQuery(qr weave.QueryRouter) {
+	NewDomainBucket().Register("domain", qr)
+	NewAccountBucket().Register("account", qr)
+}
+
+func RegisterRoutes(r weave.Registry, auth x.Authenticator) {
+	r = migration.SchemaMigratingRegistry("blueaccount", r)
+
+	domains := NewDomainBucket()
+	accounts := NewAccountBucket()
+
+	r.Handle(&RegisterDomainMsg{}, &registerDomaiHandler{
+		domains:  domains,
+		accounts: accounts,
+		auth:     auth,
+	})
+	r.Handle(&TransferDomainMsg{}, &transferDomaiHandler{
+		domains: domains,
+		auth:    auth,
+	})
+	r.Handle(&RenewDomainMsg{}, &renewDomaiHandler{
+		domains: domains,
+		auth:    auth,
+	})
+	r.Handle(&DeleteDomainMsg{}, &deleteDomaiHandler{
+		domains: domains,
+		auth:    auth,
+	})
+	r.Handle(&RegisterAccountMsg{}, &registerAccounHandler{
+		auth:     auth,
+		domains:  domains,
+		accounts: accounts,
+	})
+	r.Handle(&TransferAccountMsg{}, &transferAccounHandler{
+		auth:     auth,
+		domains:  domains,
+		accounts: accounts,
+	})
+	r.Handle(&ReplaceAccountTargetsMsg{}, &replaceAccountTargetHandler{
+		auth:     auth,
+		domains:  domains,
+		accounts: accounts,
+	})
+	r.Handle(&DeleteAccountMsg{}, &deleteAccounHandler{
+		auth:     auth,
+		domains:  domains,
+		accounts: accounts,
+	})
+
+	r.Handle(&UpdateConfigurationMsg{}, gconf.NewUpdateConfigurationHandler(
+		"blueaccount", &Configuration{}, auth))
+}
+
+type registerDomaiHandler struct {
+	auth     x.Authenticator
+	domains  orm.ModelBucket
+	accounts orm.ModelBucket
+}
+
+func (h *registerDomaiHandler) Check(ctx weave.Context, db weave.KVStore, tx weave.Tx) (*weave.CheckResult, error) {
+	if _, err := h.validate(ctx, db, tx); err != nil {
+		return nil, err
+	}
+	return &weave.CheckResult{GasAllocated: 0}, nil
+}
+
+func (h *registerDomaiHandler) Deliver(ctx weave.Context, db weave.KVStore, tx weave.Tx) (*weave.DeliverResult, error) {
+	msg, err := h.validate(ctx, db, tx)
+	if err != nil {
+		return nil, err
+	}
+
+	owner := msg.Owner
+	if len(owner) == 0 {
+		owner = x.MainSigner(ctx, h.auth).Address()
+	}
+
+	now, err := weave.BlockTime(ctx)
+	if err != nil {
+		return nil, errors.Wrap(errors.ErrState, "block time not present in context")
+	}
+	conf, err := loadConf(db)
```

Comment:
You have already run the load config in [validate stage](https://github.com/iov-one/weave/pull/1033/files#diff-87f50d45e9a324e3f8fa284ffb0a84b1R138). Why not just return not as a parameter?

Reply:
About `loadConf`, I agree with Orkun. I often return a thrid argument from `validate` (besides the properly typed message and error), if we load data there we also need in the Deliver/Check implementation. Adding a return arg seems nicer than hitting the db twice.